internal/cmd/controller/gitops/reconciler/status_controller.go

```diff
@@ -110,6 +113,28 @@ func (r *StatusReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		gitrepo.Status.Display.State = "GitUpdating"
 	}
 
+	// We're explicitly setting the ready status from a bundle here, but only if it isn't ready.
```

Comment:
Thank you for this detailed write-up :)
I wonder if all of it needs to live here, instead of being part of `setReadyStatusFromBundle`'s method-level doc, but this provides much needed context :+1:

Reply:
I find it difficult to put everything in the documentation of setReadyStatusFromBundle, as I try to describe why we can use it there the way it is being used.

Reply:
No worries, let's leave this here for now, and thanks again for writing and polishing this.